promql/parser/lex.go

```diff
@@ -568,10 +568,16 @@ Loop:
 					return lexHistogram
 				}
 				l.errorf("missing `:` for histogram descriptor")
-			} else {
-				l.errorf("bad histogram descriptor found: %q", word)
+				break Loop
 			}
-
+			// current word is Inf or NaN
```

Comment:
Nit: Use normal capitalization and punctuation.

```suggestion
			// Current word is Inf or NaN.
```